### PR TITLE
chore: Migrate samba `files_external` test from drone to github

### PR DESCRIPTION
**Pull request by susnux**
* For #41187 

## Summary
Migrate the samba files_external test from drone to github.

## TODO

- [x] merge https://github.com/nextcloud/docker-ci/pull/628 so we can use our samba image instead
- [x] @nickvergessen needs to sign the drone config

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: .github/workflows/files-external-smb.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        php-versions: ['8.0', '8.3']
+        include:
+          - php-versions: '8.0'
+            coverage: ${{ github.event_name != 'pull_request' }}
+
+    name: php${{ matrix.php-versions }}-smb
+
+    services:
+      samba:
+        image: silvershell/samba #ghcr.io/nextcloud/continuous-integration-samba:latest
```
Comment (skjnldsv) [inline .github/workflows/files-external-smb.yml:51]:
```suggestion
        image: ghcr.io/nextcloud/continuous-integration-samba:latest
```

File: .github/workflows/files-external-smb.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          ./occ app:enable --force files_external
+          echo "<?php return ['run'=>true, 'host'=>'localhost', 'user'=>'test', 'password'=>'test', 'root'=>'', 'share'=>'public'];" > apps/files_external/tests/config.smb.php
+
+      - name: Wait for smb
+        run: |
+          apps/files_external/tests/env/wait-for-connection 127.0.0.1 445 60
+
+      - name: PHPUnit
+        run: composer run test:files_external -- --verbose \
+          apps/files_external/tests/Storage/SmbTest.php \
```
Comment (come-nc) [inline .github/workflows/files-external-smb.yml:90]: This runs only this file?
Was it already the case in drone? It looks like autotest-external runs a lot more but I may miss something.

Comment (susnux) [reply .github/workflows/files-external-smb.yml:90]: The other ones are the generic `files_external` unit tests

Comment (come-nc) [reply .github/workflows/files-external-smb.yml:90]: And the drone run was not running them? Or it was but it was useless?

**Comment by nickvergessen** [comment]
Needs to be rebased after https://github.com/nextcloud/server/pull/43514 otherwise I can't sign the config

File: .github/workflows/files-external-smb.yml (hunk 2 above)
Comment (susnux) [reply .github/workflows/files-external-smb.yml:90]: Good catch - we only migrated the storage tests but not the generic ones. I will create a separate PR for that as it is unrelated to the samba migration.

**Comment by susnux** [comment]
> Needs to be rebased after https://github.com/nextcloud/server/pull/43514 otherwise I can't sign the config

@nickvergessen done

**Comment by nickvergessen** [comment]
Signed already

File: .github/workflows/files-external-smb.yml (hunk 2 above)
Comment (susnux) [reply .github/workflows/files-external-smb.yml:90]: https://github.com/nextcloud/server/actions/workflows/files-external.yml

**Review by come-nc — APPROVED** [review]

**Pull request merged**